File: src/bridge/niceaigc/examples.js
/**
 * NiceAIGC 桥接插件 - Node.js 使用示例
 */

import NiceAIGCBridge from './bridge.js';

// ==================== 示例 1: 基础使用 ====================

async function exampleBasic() {
  console.log('='.repeat(50));
  console.log('示例 1: 基础使用');
  console.log('='.repeat(50));
  
  const bridge = new NiceAIGCBridge();
  
  const message = '请帮我润色以下论文摘要：\n\n本研究探讨了气候变化对生态系统的影响...';
  
  const response = await bridge.sendMessage(message);
  
  console.log(`\nAI 响应:\n${response}`);
  return response;
}

// ==================== 示例 2: 保存结果到文件 ====================

async function exampleSaveToFile() {
  console.log('\n' + '='.repeat(50));
  console.log('示例 2: 保存结果到文件');
  console.log('='.repeat(50));
  
  const bridge = new NiceAIGCBridge();
  
  const message = '请列出 5 个论文写作中常见的逻辑错误';
  
  const response = await bridge.sendAndSave(message, '/tmp/niceaigc_output.txt');
  
  console.log(`\n结果已保存到 /tmp/niceaigc_output.txt`);
  console.log(`响应预览:\n${response.substring(0, 200)}...`);
  return response;
}

// ==================== 示例 3: 集成到 Express 后端 ====================

function exampleExpressIntegration() {
  console.log('\n' + '='.repeat(50));
  console.log('示例 3: Express 后端集成');
  console.log('='.repeat(50));
  
  const code = `
// app.js - 你的 Express 后端
import express from 'express';
import NiceAIGCBridge from './niceaigc-bridge/bridge.js';

const app = express();
app.use(express.json());

const bridge = new NiceAIGCBridge();

// API 端点：调用 NiceAIGC
app.post('/api/ai/polish', async (req, res) => {
  const { text } = req.body;
  
  if (!text) {
    return res.status(400).json({ error: '缺少 text 参数' });
  }
  
  try {
    const message = \`请帮我润色以下学术文本，使其更加专业和流畅：\\n\\n\${text}\`;
    const response = await bridge.sendMessage(message);
    
    res.json({
      success: true,
      original: text,
      polished: response
    });
  } catch (error) {
    res.status(500).json({
      success: false,
      error: error.message
    });
  }
});

// 启动服务
app.listen(3000, () => {
  console.log('后端服务运行在 http://localhost:3000');
});
`;
  
  console.log(code);
  return code;
}

// ==================== 示例 4: 前端调用示例 ====================

function exampleFrontendCall() {
  console.log('\n' + '='.repeat(50));
  console.log('示例 4: 前端调用示例');
  console.log('='.repeat(50));
  
  const code = `
// 前端 JavaScript（浏览器环境）
async function polishText(text) {
  const response = await fetch('http://localhost:8765/chat', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json'
    },
    body: JSON.stringify({
      message: \`请帮我润色以下学术文本：\\n\\n\${text}\`
    })
  });
  
  const data = await response.json();
  
  if (data.success) {
    return data.response;
  } else {
    throw new Error(data.error);
  }
}

// 使用示例
const original = "This study is about climate change.";
const polished = await polishText(original);
console.log('润色后:', polished);
`;
  
  console.log(code);
  return code;
}

// ==================== 示例 5: 批量处理 ====================

async function exampleBatchProcessing() {
  console.log('\n' + '='.repeat(50));
  console.log('示例 5: 批量处理');
  console.log('='.repeat(50));
  
  const bridge = new NiceAIGCBridge();
  
  const messages = [
    '润色：This study is about climate change.',
    '润色：We found some interesting results.',
    '润色：The method we used is good.',
  ];
  
  console.log('批量处理 3 条消息...\\n');
  
  for (let i = 0; i < messages.length; i++) {
    const msg = messages[i];
    console.log(\`[\${i + 1}/\${messages.length}] \${msg}\`);
    
    try {
      const response = await bridge.sendMessage(msg);
      console.log(\`    → \${response.substring(0, 100)}...\\n\`);
    } catch (error) {
      console.log(\`    → 错误：\${error.message}\\n\`);
    }
    
    // 延迟避免请求过快
    if (i < messages.length - 1) {
      await bridge.sleep(2000);
    }
  }
}

// ==================== 示例 6: 测试连接 ====================

async function exampleTestConnection() {
  console.log('\n' + '='.repeat(50));
  console.log('示例 6: 测试连接');
  console.log('='.repeat(50));
  
  const bridge = new NiceAIGCBridge();
  
  const ok = await bridge.testConnection();
  
  if (ok) {
    console.log('\\n✅ 连接测试成功！');
  } else {
    console.log('\\n❌ 连接测试失败！');
  }
  
  return ok;
}

// ==================== 主程序 ====================

async function main() {
  console.log('\\n' + '='.repeat(60));
  console.log('   NiceAIGC 桥接插件 - Node.js 使用示例');
  console.log('='.repeat(60));
  
  const examples = [
    { name: '基础使用', fn: exampleBasic },
    { name: '保存文件', fn: exampleSaveToFile },
    { name: 'Express 集成', fn: exampleExpressIntegration },
    { name: '前端调用', fn: exampleFrontendCall },
    { name: '批量处理', fn: exampleBatchProcessing },
    { name: '测试连接', fn: exampleTestConnection },
  ];
  
  if (process.argv.length > 2) {
    // 运行指定示例
    const idx = parseInt(process.argv[2]) - 1;
    if (idx >= 0 && idx < examples.length) {
      await examples[idx].fn();
    } else {
      console.log(\`无效示例编号：\${idx + 1}\`);
    }
  } else {
    // 显示菜单
    console.log('\\n可用示例：');
    examples.forEach((ex, i) => {
      console.log(\`  \${i + 1}. \${ex.name}\`);
    });
    console.log('\\n运行方式：node examples.js [示例编号]');
    console.log('例如：node examples.js 1');
  }
}

main().catch(console.error);

File: src/bridge/chat-bridge/examples.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
NiceAIGC 桥接插件 - 使用示例

展示如何将桥接插件集成到学术论文写作项目中
"""

import sys
import json
from pathlib import Path

# 添加桥接插件到路径
sys.path.insert(0, str(Path(__file__).parent))

from bridge import NiceAIGCBridge


# ==================== 示例 1: 基础使用 ====================

def example_basic():
    """基础使用示例"""
    print("=" * 50)
    print("示例 1: 基础使用")
    print("=" * 50)
    
    bridge = NiceAIGCBridge()
    
    message = "请帮我润色以下论文摘要：\n\n本研究探讨了气候变化对生态系统的影响..."
    
    response = bridge.send_message(message)
    
    print(f"\nAI 响应:\n{response}")
    return response


# ==================== 示例 2: 指定模式 ====================

def example_with_mode():
    """指定运行模式"""
    print("\n" + "=" * 50)
    print("示例 2: 指定模式")
    print("=" * 50)
    
    bridge = NiceAIGCBridge()
    
    # 强制使用浏览器模式
    message = "帮我改写这句话，使其更学术化：The results show that..."
    
    response = bridge.send_message(message, mode='browser')
    
    print(f"\nAI 响应:\n{response}")
    return response


# ==================== 示例 3: 保存结果到文件 ====================

def example_save_to_file():
    """保存结果到文件"""
    print("\n" + "=" * 50)
    print("示例 3: 保存结果到文件")
    print("=" * 50)
    
    bridge = NiceAIGCBridge()
    
    message = "请列出 5 个论文写作中常见的逻辑错误"
    
    response = bridge.send_and_save(
        message,
        output_path='/tmp/niceaigc_output.txt'
    )
    
    print(f"\n结果已保存到 /tmp/niceaigc_output.txt")
    print(f"响应预览:\n{response[:200]}...")
    return response


# ==================== 示例 4: 集成到 FastAPI 后端 ====================

def example_fastapi_integration():
    """FastAPI 集成示例（伪代码）"""
    print("\n" + "=" * 50)
    print("示例 4: FastAPI 集成")
    print("=" * 50)
    
    fastapi_code = '''
from fastapi import FastAPI
from pydantic import BaseModel
from niceaigc_bridge import NiceAIGCBridge

app = FastAPI()
bridge = NiceAIGCBridge()

class ChatRequest(BaseModel):
    message: str
    mode: str = "auto"

@app.post("/api/chat")
async def chat(request: ChatRequest):
    """处理聊天请求"""
    response = bridge.send_message(request.message, mode=request.mode)
    return {"response": response}

@app.post("/api/chat/file")
async def chat_with_file(request: ChatRequest):
    """处理带文件输出的聊天请求"""
    response = bridge.send_and_save(request.message)
    return {"response": response, "saved_to": "/tmp/niceaigc_response.txt"}

# 运行：uvicorn main:app --reload --port 8000
'''
    
    print(fastapi_code)
    return fastapi_code


# ==================== 示例 5: 多轮对话 ====================

def example_conversation():
    """多轮对话示例"""
    print("\n" + "=" * 50)
    print("示例 5: 多轮对话")
    print("=" * 50)
    
    bridge = NiceAIGCBridge()
    
    # 从 API 模式模块导入（支持多轮对话）
    from api_mode import NiceAIGCAPI
    
    # 加载配置
    config_path = Path(__file__).parent / 'config.json'
    with open(config_path, 'r', encoding='utf-8') as f:
        config = json.load(f)
    
    api = NiceAIGCAPI(config)
    
    # 构建对话历史
    messages = [
        {'role': 'system', 'content': '你是一位学术论文编辑专家'},
        {'role': 'user', 'content': '帮我检查这句话的语法：The data was analyzed by using SPSS.'},
    ]
    
    try:
        response = api.send_conversation(messages)
        print(f"\nAI 响应:\n{response}")
        
        # 继续对话
        messages.append({'role': 'assistant', 'content': response})
        messages.append({'role': 'user', 'content': '还有其他需要注意的地方吗？'})
        
        response2 = api.send_conversation(messages)
        print(f"\n继续对话:\n{response2}")
        
    except Exception as e:
        print(f"多轮对话需要 API 模式支持：{e}")
        print("浏览器模式暂不支持多轮对话上下文")


# ==================== 示例 6: 批量处理 ====================

def example_batch_processing():
    """批量处理示例"""
    print("\n" + "=" * 50)
    print("示例 6: 批量处理")
    print("=" * 50)
    
    bridge = NiceAIGCBridge()
    
    messages = [
        "润色：This study is about climate change.",
        "润色：We found some interesting results.",
        "润色：The method we used is good.",
    ]
    
    print("批量处理 3 条消息...\n")
    
    for i, msg in enumerate(messages, 1):
        print(f"[{i}/3] {msg}")
        try:
            response = bridge.send_message(msg)
            print(f"    → {response[:100]}...\n")
        except Exception as e:
            print(f"    → 错误：{e}\n")


# ==================== 主程序 ====================

if __name__ == '__main__':
    print("\n" + "=" * 60)
    print("   NiceAIGC 桥接插件 - 使用示例")
    print("=" * 60)
    
    examples = [
        ("基础使用", example_basic),
        ("指定模式", example_with_mode),
        ("保存文件", example_save_to_file),
        ("FastAPI 集成", example_fastapi_integration),
        ("多轮对话", example_conversation),
        ("批量处理", example_batch_processing),
    ]
    
    if len(sys.argv) > 1:
        # 运行指定示例
        try:
            idx = int(sys.argv[1]) - 1
            if 0 <= idx < len(examples):
                examples[idx][1]()
            else:
                print(f"无效示例编号：{idx + 1}")
        except ValueError:
            print("请使用数字选择示例（1-6）")
    else:
        # 显示菜单
        print("\n可用示例：")
        for i, (name, _) in enumerate(examples, 1):
            print(f"  {i}. {name}")
        print("\n运行方式：python3 examples.py [示例编号]")
        print("例如：python3 examples.py 1")
